File: LoggingiOS/LoggingiOS/auth/AuthViewModel.swift
import Foundation

class AuthViewModel {
    private let service = AuthService()

    func login(username: String, password: String) -> Bool {
        return service.login(username: username, password: password)
    }

    func signup(username: String, password: String) -> Bool {
        return service.signup(username: username, password: password)
    }
}

File: Logging/app/src/main/java/com/example/logging/auth/AuthViewModel.kt
package com.example.logging.auth

import androidx.lifecycle.ViewModel

class AuthViewModel : ViewModel() {
    private val repository = AuthRepository()

    fun login(username: String, password: String): Boolean {
        return repository.login(username, password)
    }

    fun signup(username: String, password: String): Boolean {
        return repository.signup(username, password)
    }
}
